### ISSUE TITLE
leaflet with dioxus

### ISSUE THREAD

**Issue by AntoineRenaud91** [slowtec/leaflet-rs#30]
Hi,
I'm seeking assistance with using Leaflet in the Dioxus framework. Below is my component. The map container is visible, but the tile layer and marker are not. Do you have any suggestions as to why this might be happening?

````rust
use dioxus::prelude::*;
use gloo_utils::document;
use leaflet::{Map,MapOptions,TileLayer,Marker, LatLng};
use web_sys::{wasm_bindgen::JsCast, HtmlElement};

fn add_map_to_container(id: &str) {
    let container:HtmlElement = document().get_element_by_id(id)
        .unwrap()
        .dyn_into()
        .unwrap();
    container.set_class_name("map");
    let map = Map::new_with_element(&container,&MapOptions::default());
    TileLayer::new("https://tile.openstreetmap.org/{z}/{x}/{y}.png").add_to(&map);
    Marker::new(&LatLng::new(0., 0.)).add_to(&map);
}

#[inline_props]
pub fn MapContainer(cx: Scope) -> Element {
    render! {
        div {
            id: "map",
            width: "100%",
            height: "100%",
            onmounted: move |_| add_map_to_container("map")
        }
    }
}
````

**Comment by s1gtrap** [comment]
On the off chance you're still looking for assistance 2/3 of a year later, I'm pretty sure it was a matter of you forgetting to call `setView`. Attempting to drag the container causes the error `Uncaught Error: Set map center and zoom first.` so I think it's genuinely just a matter of adding:
```rust
map.set_view(&LatLng::new(0.0, 0.0), 0.0);
```

Here's a working example for Dioxus 0.5:
```rust
#![allow(non_snake_case)]

use dioxus::prelude::*;
use dioxus_logger::tracing::{info, Level};
use gloo_utils::document;
use leaflet::{LatLng, Map, MapOptions, Marker, TileLayer};
use web_sys::{wasm_bindgen::JsCast, HtmlElement};

fn add_map_to_container(id: &str) {
    let container: HtmlElement = document()
        .get_element_by_id(id)
        .unwrap()
        .dyn_into()
        .unwrap();
    container.set_class_name("map");
    let map = Map::new_with_element(&container, &MapOptions::default());
    TileLayer::new("https://tile.openstreetmap.org/{z}/{x}/{y}.png").add_to(&map);
    map.set_view(&LatLng::new(0.0, 0.0), 0.0);
    Marker::new(&LatLng::new(0., 0.)).add_to(&map);
}

#[component]
pub fn MapContainer() -> Element {
    rsx! {
        div {
            id: "map",
            width: "100%",
            height: "100vh",
            onmounted: move |_| add_map_to_container("map")
        }
    }
}

#[component]
fn App() -> Element {
    rsx! {
    MapContainer {}
        }
}

fn main() {
    // Init logger
    dioxus_logger::init(Level::INFO).expect("failed to init logger");
    info!("starting app");
    launch(App);
}
```
Also don't forget to load Leaflets styles/scripts:
```toml
[application]

# App (Project) Name
name = "leaflet-dioxus"

# Dioxus App Default Platform
# web, desktop, fullstack
default_platform = "web"

# `build` & `serve` dist path
out_dir = "dist"

# resource (assets) file folder
asset_dir = "assets"

[web.app]

# HTML title tag content
title = "leaflet-dioxus"

[web.watcher]

# when watcher trigger, regenerate the `index.html`
reload_html = true

# which files or dirs will be watcher monitoring
watch_path = ["src", "assets"]

# include `assets` in web platform
[web.resource]

# CSS style file

style = ["https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.9.4/leaflet.min.css"]

# Javascript code file
script = ["https://cdnjs.cloudflare.com/ajax/libs/leaflet/1.9.4/leaflet.min.js"]

[web.resource.dev]

# Javascript code file
# serve: [dev-server] only
script = []
```

**Comment by flosse** [comment]
Solved by #33  and https://github.com/slowtec/leaflet-rs/issues/30#issuecomment-2269815881 thanks to @s1gtrap :+1:
